account_move_base_import/models/account_move.py

```diff
@@ -0,0 +1,347 @@
+                raise ErrorTooManyPartner(
+                    _('Line named "%s" was matched by more than one '
+                      'partner while looking on partner by name') %
+                    line.name)
+            res['partner_id'] = result[0][0]
+        return res
+
+
+class AccountMoveLine(models.Model):
+    """
+    Add sparse field on the statement line to allow to store all the bank infos
+    that are given by a bank/office. You can then add you own in your module.
+    The idea here is to store all bank/office infos in the
+    additional_bank_fields serialized field when importing the file. If many
+    values, add a tab in the bank statement line to store your specific one.
+    Have a look in account_move_base_import module to see how we've done
+    it.
+    """
+    _inherit = "account.move.line"
+    _order = "already_completed desc, date asc"
+
+    already_completed = fields.Boolean(
+        string="Auto-Completed",
+        default=False,
+        help="When this checkbox is ticked, the auto-completion "
+        "process/button will ignore this line.")
+
+    def _get_line_values_from_rules(self, rules):
+        """We'll try to find out the values related to the line based on rules
+        set on the profile.. We will ignore line for which already_completed
+        is ticked.
+
+        :return:
+            A dict of dict value that can be passed directly to the write
+            method of the statement line or {}. The first dict has statement
+            line ID as a key: {117009: {'partner_id': 100997,
+            'account_id': 489L}}
+        """
+        self.ensure_one()
+        vals = {}
+        if not self.already_completed:
+            # Ask the rule
+            vals = self.env['account.journal']._find_values_from_rules(
+                rules, self
+            )
+        return vals
+
+
+class AccountMove(models.Model):
+    """We add a basic button and stuff to support the auto-completion
+    of the bank statement once line have been imported or manually fulfill.
+    """
+    _name = 'account.move'
+    _inherit = ['account.move', 'mail.thread']
+
+    used_for_completion = fields.Boolean(
+        related='journal_id.used_for_completion',
+        readonly=True)
+    completion_logs = fields.Text(string='Completion Log', readonly=True)
+    # partner_id is a native field of the account module
+    # (related='line_ids.partner_id', store=True, readonly=True)
+    partner_id = fields.Many2one(related=False, compute='_compute_partner_id')
+    import_partner_id = fields.Many2one('res.partner',
+                                        string="Partner from import")
+
+    @api.depends('line_ids.partner_id', 'import_partner_id')
+    def _compute_partner_id(self):
+        for move in self:
+            if move.import_partner_id:
+                move.partner_id = move.import_partner_id
+            elif move.line_ids:
+                move.partner_id = move.line_ids[0].partner_id
```

Comment:
doesn't this change default behavior of odoo? move can have different partners in this case we should provide same check as https://github.com/odoo/odoo/blob/1bd434da34731b5467cc17fdffe6f3ed5dfea2d4/addons/account/models/account_move.py#L77